### DISCUSSION TITLE
Error in RunPCA: "You're computing too large a percentage of total singular values, use a standard svd instead.did not converge--results might be invalid!; try increasing work or max it"

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [satijalab/seurat#4567] [Q&A] [answered]
1. Following up on #3945, would anyone mind helping me to explain the error/warning shown below when I run runPCA on my seurat object? And, whether I should be concerned about it? The object was SCTransform()-ed as below, so I thought all features should have been scaled.

2. Before and after regressing out cc.difference looks the same but now the PC cordinates go towards more negative values. Does it look right? 

Thank you very much for your help!

```
# Normalize by SCTransform without scaling nor centering
object <- SCTransform(object, verbose = FALSE, do.scale=FALSE, do.center=FALSE, method = "glmGamPoi")

object <- CellCycleScoring(object, s.features = s.genes, g2m.features = g2m.genes, set.ident = FALSE)

object$CC.Difference <- object$S.Score - object$G2M.Score

# Regress percent.mt and CC.difference on RNA slots
DefaultAssay(object) <- 'RNA'
object <- SCTransform(object, vars.to.regress = c("percent.mt","CC.Difference"), verbose = FALSE, do.scale=TRUE, do.center=TRUE)

DefaultAssay(object) <- 'SCT'
# Error/warning appears at this step
object <- RunPCA(object, features = c(s.genes, g2m.genes))

```

<img width="760" alt="Screen Shot 2021-06-03 at 11 21 05 AM" src="https://user-images.githubusercontent.com/62179082/120629507-cb7e8080-c45d-11eb-94f6-da6393561839.png">

**Before regression of cc.difference:**
```
object <- RunPCA(object, features = c(s.genes, g2m.genes))
DimPlot(object, group.by = "Phase", order = T)
```
<img width="521" alt="Screen Shot 2021-06-03 at 11 24 38 AM" src="https://user-images.githubusercontent.com/62179082/120629992-49db2280-c45e-11eb-8901-d51431cbdeb5.png">

**After regression of cc.difference as above:**
```
object <- RunPCA(object, features = c(s.genes, g2m.genes))
DimPlot(cxcr5posnegcd8_filtergene_mergedDonors_cellselect, order = T, group.by = "orig.ident")
```

<img width="525" alt="Screen Shot 2021-06-03 at 11 27 13 AM" src="https://user-images.githubusercontent.com/62179082/120630335-a63e4200-c45e-11eb-88b6-56337d464eb5.png">

**Comment by timoast** [comment, marked as answer]
Why are you only using the cell cycle genes to run PCA here?

You see this warning because you're trying to compute the top 50 PCs using only 35 genes as input (which is impossible). The warning tells you that it's instead computing the top 35 PCs.

**Reply by denvercal1234GitHub** [reply]
Thank you for your reply, @timoast!

I followed the tutorial (https://satijalab.org/seurat/articles/cell_cycle_vignette.html), using cc genes in PCA after regressing it out as shown in the code above to check whether the cc genes would still cluster the cells after regressing. But it looks like my cells were not affected by the cc genes anyway (as shown in the first DimPlot above). 

Am I misunderstanding something here?

Thanks for your help.

**Reply by timoast** [reply]
In the vignette, PCA is only run on the CC genes to show that the resulting PCs capture cell cycle, it's not used for downstream analysis and there's not really any reason to do this.

**Reply by denvercal1234GitHub** [reply]
Thank you for your response. But I think it does say that running PCA on the cc genes and plot the PCs after regressing in that dataset now show reduction of the impact of cc genes on the cell clustering (below). That is why I did it just to check the impact of the regressing. I then did RunPCA on variable genes for downstream analysis after that (I assume running second time the RunPCA will override the first).

I guess my question is how to check whether the regressing works, besides just running RunPCA on variable genes and check whether cc genes are still in the first few PCs? 

<img width="790" alt="Screen Shot 2021-06-05 at 12 12 30 AM" src="https://user-images.githubusercontent.com/62179082/120871130-b9eac500-c592-11eb-8c27-bb858024d096.png">

**Reply by timoast** [reply]
One of your original questions was what is the source of the warning in `RunPCA`, so that's why the warning occurs. If cell cycle is not a significant source of variation in your dataset, then there's not really any need to regress out cell cycle. Typically you would first run an analysis without doing any cell cycle regression, and then see if you're getting clusters based on cell cycle stage.

**Reply by denvercal1234GitHub** [reply]
Thank you Tim for your advice!
